### PR TITLE
Add major alarm as separate MotionNotReadyCode value

### PR DESCRIPTION
**Pull request by jimmy-mcelwain**
Prerequisite for [motoros2 pr](https://github.com/Yaskawa-Global/motoros2/pull/298) to be merged (as is)

### CODE REVIEW

**Review by ted-miller — DISMISSED** [review]
Will merge with https://github.com/Yaskawa-Global/motoros2/pull/298

**Review by ted-miller — CHANGES_REQUESTED** [review]
Just noticed that this conflicts with #17. We'll need to rebase after merging that.

File: msg/MotionReadyEnum.msg [hunk 1]
```diff
@@ -61,4 +61,7 @@ string NOT_READY_OTHER_TRAJ_MODE_ACTIVE_STR="Another trajectory mode is already
 uint16 NOT_READY_NOT_CONT_CYCLE_MODE=115
 string NOT_READY_NOT_CONT_CYCLE_MODE_STR="Continuous cycle mode not set. Please set the cycle mode to continuous"
 
+uint16 NOT_READY_MAJOR_ALARM=116
+string NOT_READY_MAJOR_ALARM_STR="The controller has an active Major Alarm that cannot be reset. Check teach pendant"
```
Comment (gavanderhoorn) [inline msg/MotionReadyEnum.msg:65]: Let's be kind :)

```suggestion
string NOT_READY_MAJOR_ALARM_STR="The controller has an active Major Alarm that cannot be reset. Please check the teach pendant"
```

PS: isn't it technically the *programming* pendant?

**Review by gavanderhoorn — CHANGES_REQUESTED** [review]

File: msg/MotionReadyEnum.msg (hunk 1 above)
Comment (ted-miller) [reply msg/MotionReadyEnum.msg:65]: > Let's be kind :)

NEIN! You vill check zee pendant! Do not dilly dally.

> PS: isn't it technically the programming pendant?

Meh... interchangeable terminology.

Comment (jimmy-mcelwain) [reply msg/MotionReadyEnum.msg]: Updated to be kinder 😄 

I kept it as "teach pendant" to be consistent with other `MotionNotReadyCode` strings.

File: msg/MotionReadyEnum.msg [hunk 2]
```diff
@@ -61,4 +61,7 @@ string NOT_READY_OTHER_TRAJ_MODE_ACTIVE_STR="Another trajectory mode is already
 uint16 NOT_READY_NOT_CONT_CYCLE_MODE=115
 string NOT_READY_NOT_CONT_CYCLE_MODE_STR="Continuous cycle mode not set. Please set the cycle mode to continuous"
 
+uint16 NOT_READY_MAJOR_ALARM=116
+string NOT_READY_MAJOR_ALARM_STR="The controller has an active Major Alarm that cannot be reset. Please check the teach pendant"
```
Comment (gavanderhoorn) [inline msg/MotionReadyEnum.msg:65]:
```suggestion
string NOT_READY_MAJOR_ALARM_STR="Major Alarms can't be reset remotely. Please check the teach pendant"
```

**Comment by gavanderhoorn** [comment]
@ted-miller wrote:
> conflict with #17

while this is true, I'd suggest we merge this one first, as it's a relatively simple change and the corresponding https://github.com/Yaskawa-Global/motoros2/pull/298 was just merged.

File: msg/MotionReadyEnum.msg (hunk 2 above)
Comment (jimmy-mcelwain) [reply msg/MotionReadyEnum.msg:65]: That suggested message is good as a message for if `/reset_error` is called, but as a typical `MotionNotReadyCode` string it doesn't exactly do its job of telling the user what is preventing motion. For example if `Ros_Controller_GetNotReadySubcode()` is called during `Ros_MotionControl_StartMotionMode()`, the corresponding message doesn't really fit.

Comment (gavanderhoorn) [reply msg/MotionReadyEnum.msg:65]: Ok, what about:

```suggestion
string NOT_READY_MAJOR_ALARM_STR="Major Alarms can't be reset (remotely). Please check the teach pendant"
```

Comment (jimmy-mcelwain) [reply msg/MotionReadyEnum.msg:65]: I just think that if I call `/start_traj_mode` and I get that message in response, it is a little confusing

**Review by ted-miller — APPROVED** [review]

**Pull request merged**